### PR TITLE
core(config): remove config.artifacts; always use auditMode

### PR DESCRIPTION
**Pull request by brendankenny**
part of #4333

remove support for artifacts loaded in the config and the kind of wonky way it loaded traces and devtoolsLogs from there. Now the official way to load artifacts and skip the gather step is with `--audit-mode`/`auditMode` in the config.

Required creating some artifacts fixtures to port over old tests that loaded with `config.artifacts`, but on the plus side it means more tests for `--audit-mode`.

### CODE REVIEW

File: lighthouse-core/runner.js [hunk 1]
```diff
@@ -115,7 +113,8 @@ class Runner {
       const resultsById = {};
       for (const audit of auditResults) resultsById[audit.name] = audit;
 
-      let reportCategories;
+      /** @type {Array<LH.Result.Category>} */
+      let reportCategories = [];
```
Comment (brendankenny) [inline lighthouse-core/runner.js]: part of the discussed change to default to having an empty `categories` array, even if none in the config (we could maybe start defaulting to `[]` in the config, but we'll see)

File: lighthouse-core/test/fixtures/artifacts/perflog/defaultPass.trace.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[
```
Comment (brendankenny) [inline lighthouse-core/test/fixtures/artifacts/perflog/defaultPass.trace.json]: copied over from `fixtures/traces/trace-user-timings.json` cause it was small and we really only need to test the loading part :)

File: lighthouse-core/test/gather/fake-driver.js [hunk 3]
```diff
@@ -60,7 +60,7 @@ module.exports = {
   },
   beginDevtoolsLog() {},
   endDevtoolsLog() {
-    return require('../fixtures/perflog.json');
+    return require('../fixtures/artifacts/perflog/defaultPass.devtoolslog.json');
```
Comment (brendankenny) [inline lighthouse-core/test/gather/fake-driver.js]: this was big, so just moved over to the artifacts fixtures and moved the two tests using it to point at the new path

File: lighthouse-core/test/index-test.js [hunk 4]
```diff
@@ -87,18 +87,17 @@ describe('Module Tests', function() {
       });
   });
 
-  it.skip('should return formatted audit results when given no categories', function() {
+  it.only('should return formatted LHR when given no categories', function() {
```
Comment (brendankenny) [inline lighthouse-core/test/index-test.js]: this was skipped when `-GA` was added. Reenabled :P

File: lighthouse-core/test/runner-test.js [hunk 5]
```diff
@@ -148,25 +148,6 @@ describe('Runner', () => {
       });
   });
 
-  it('accepts existing artifacts', () => {
```
Comment (brendankenny) [inline lighthouse-core/test/runner-test.js]: overlap with lots of other -GA and `auditMode` tests, so removed

File: lighthouse-core/test/index-test.js [hunk 6]
```diff
@@ -87,18 +87,17 @@ describe('Module Tests', function() {
       });
   });
 
-  it.skip('should return formatted audit results when given no categories', function() {
+  it('should return formatted LHR when given no categories', function() {
```
Comment (brendankenny) [inline lighthouse-core/test/index-test.js]: this was skipped when `-GA` was added. Reenabled :P

File: lighthouse-core/config/config.js [hunk 7]
```diff
@@ -17,89 +17,6 @@ const path = require('path');
 const Audit = require('../audits/audit');
 const Runner = require('../runner');
 
-// cleanTrace is run to remove duplicate TracingStartedInPage events,
-// and to change TracingStartedInBrowser events into TracingStartedInPage.
-// This is done by searching for most occuring threads and basing new events
-// off of those.
-function cleanTrace(trace) {
```
[the remark this answers is not in the document]
Comment (paulirish) [reply lighthouse-core/config/config.js]: aye. this removes support for traces captured by webdriver, but at this point, i'm just fine with that. :)

**Pull request merged**